### DISCUSSION TITLE
Using useIsFetching to force component's rerender

### DISCUSSION THREAD

**Discussion by m-fraczek** [TanStack/query#7071] [Q&A] [answered]
Hey, I'm trying to access Query's data from a different component with subscription without reusing useQuery. I've been using queryClient.getQueryState to access data and useIsFetching to rerender component on each refetch, I'm wondering if this is the correct approach? I cannot just use useQuery again since I cannot access required arguments from every component. Thanks in advance

**Comment by TkDodo** [comment, marked as answer]
how can you call `getQueryState` if you don't know the `queryKey` ? If you know the `queryKey`, you can just as well call `useQuery`.
